pkg/operator/ceph/cluster/cleanup.go

```diff
@@ -85,16 +89,21 @@ func (c *ClusterController) startCleanUpJobs(cluster *cephv1.CephCluster, cephHo
 	}
 }
 
-func (c *ClusterController) cleanUpJobContainer(cluster *cephv1.CephCluster, monSecret string) v1.Container {
+func (c *ClusterController) cleanUpJobContainer(cluster *cephv1.CephCluster, monSecret, cephFSID string) v1.Container {
 	volumeMounts := []v1.VolumeMount{}
 	envVars := []v1.EnvVar{}
 	if cluster.Spec.CleanupPolicy.DeleteDataDirOnHosts != "" && cluster.Spec.DataDirHostPath != "" {
 		hostPathVolumeMount := v1.VolumeMount{Name: volumeName, MountPath: cluster.Spec.DataDirHostPath}
+		devMount := v1.VolumeMount{Name: "devices", MountPath: "/dev"}
 		volumeMounts = append(volumeMounts, hostPathVolumeMount)
+		volumeMounts = append(volumeMounts, devMount)
 		envVars = append(envVars, []v1.EnvVar{
 			{Name: dataDirHostPath, Value: cluster.Spec.DataDirHostPath},
 			{Name: namespaceDir, Value: cluster.Namespace},
 			{Name: monitorSecret, Value: monSecret},
+			{Name: clusterFSID, Value: cephFSID},
+			{Name: clusterName, Value: cluster.Name},
+			{Name: clusterOnPVC, Value: strconv.FormatBool(len(cluster.Spec.Storage.StorageClassDeviceSets) != 0)},
```

Comment:
Theoretically someone could have both types of OSDs, both on PVC and not. Seems like we should just attempt to clean up all the OSDs in the cluster that we find on the node, whether or not they are on PVC. Just discover them all, right?

Reply:
In practice, I agree but the thing is we have to have this so we can different which LVM filter to apply. If we don't have the correct filter we will be missing some OSDs. That's how `UpdateLVMConfig()` works today.

Reply:
What if we searched for both types of OSDs? For example, set the LVM config for non-PVCs and purge the OSDs, then update the LVM config for PVCs and detect the other OSDs. Then if there is a mix because of upgrade from different releases or different settings in the CR, we would purge both types.

Reply:
Not needed anymore.